src/v/cluster/health_monitor_types.h

```diff
@@ -80,11 +81,12 @@ struct topic_status {
  * instance of time
  */
 struct node_health_report {
```

Comment:
The compat backport for 21.11.12 (cc220238) will still assert the version to be `<= 1` 

Maybe encode version 2 only if maintenance_mode feature is active?

Reply:
it's true, but isn't this true for nearly every mixed version interaction we have? old node receives message from newer node is and crashes?

Reply:
Kinda, but I think for a lot of stuff it wasn't as acute, if a particular message type only comes up under certain circumstances -- the health messages are kind of an insta-kill even on an idle cluster.

Reply:
> but I think for a lot of stuff it wasn't as acute,

oh of course, this is an active pull from the nodes every few seconds. will adjust soon

Reply:
Fixed